addOns/client/src/main/java/org/zaproxy/addon/client/ClientIntegrationAPI.java

```diff
@@ -152,7 +152,9 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
                     if (!Files.isWritable(path.getParent()) || !exportPath.endsWith(".yaml")) {
                         throw new IllegalArgumentException("Invalid export path: " + exportPath);
                     }
-                    this.extension.exportClientMap(exportPath);
+                    if (!this.extension.exportClientMap(exportPath)) {
+                        throw new RuntimeException("Failed to export client map: " + exportPath);
```

Comment:
This should be an ApiException similar to the default case below. The Type wouldn't be BAD_ACTION, probably INTERNAL_ERROR (@thc202 / @psiinon ?)

Reply:
Worth checking what other similar API endpoints use - they should be consistent.